### PR TITLE
[189201] Enable embedded mode

### PR DESCRIPTION
**Pull request by aDaichiOta**
https://acsmine.tok.access-company.com/redmine/issues/189201

### CODE REVIEW

File: core/version/gear.ex [hunk 1]
```diff
@@ -29,15 +29,17 @@ defmodule AntikytheraCore.Version.Gear do
...
-          if not skip_to_load? do
-            case :code.load_file(mod) do
-              {:module, _}      -> :ok
-              {:error , reason} -> raise "Failed to load '#{mod}': #{reason}"
-            end
+          case :code.load_file(mod) do
+            {:module, _}      -> :ok
+            {:error , reason} ->
+              # Croma defines modules (e.g. `Elixir.Croma.TypeGen.Nilable.Antikythera.Email`) automatically in the gear.
+              # As a result, some modules which have the same name are defined in two or more gears, and loading these modules causes an error.
```
Comment (irisTa56) [inline core/version/gear.ex]: (optional)

```suggestion
              # As a result, some modules with the same name are defined in two or more gears, and loading these modules causes an error.
```

I think this is easier to read.

File: core/version/gear.ex [hunk 2]
```diff
@@ -32,12 +32,18 @@ defmodule AntikytheraCore.Version.Gear do
...
+                {:module, _}           -> :ok
+                # When three or more same-name modules are loaded at the same time,
+                # `:code.load_file(mod)` returns `:not_purged`. We can safely ignore the error in that case.
+                {:error , :not_purged} -> :ok
+                {:error , reason}      -> raise "Failed to load '#{mod}': #{reason}"
+              end
             end
+          else
+            {:module, _} = :code.load_file(mod)
           end
```
Comment (irisTa56) [inline core/version/gear.ex]: (optional)
I think writing shorter case first is more readable.

```elixir
          if not auto_generated_module?(mod) do
            {:module, _} = :code.load_file(mod)
          else
            if !:erlang.module_loaded(mod) do
              case :code.load_file(mod) do
                {:module, _}           -> :ok
                # When three or more same-name modules are loaded at the same time,
                # `:code.load_file(mod)` returns `:not_purged`. We can safely ignore the error in that case.
                {:error , :not_purged} -> :ok
                {:error , reason}      -> raise "Failed to load '#{mod}': #{reason}"
              end
            end
          end
```

**Comment by irisTa56** [comment]
Just a comment.
LGTM!

File: core/version/gear.ex (hunk 2 above)
Comment (aDaichiOta) [reply core/version/gear.ex]: Fixed be13b4df2c711637193077408f3060fea4c9efc4

**Comment by ttokit** [comment]
LGTM!

**Comment by aMasakiTakahashi** [comment]
LGTM!

**Pull request merged**